samtranslator/model/__init__.py

```diff
@@ -345,15 +346,15 @@ def validate_properties_and_return_model(self, cls: Type[RT]) -> RT:
         """
         try:
             return cls.parse_obj(self._generate_resource_dict()["Properties"])
-        except pydantic.error_wrappers.ValidationError as e:
-            error_properties: str = ""
-            with suppress(KeyError):
-                error_properties = ".".join(str(x) for x in e.errors()[0]["loc"])
-            raise InvalidResourceException(self.logical_id, f"Property '{error_properties}' is invalid.") from e
-        except AttributeError as e:
-            raise InvalidResourceException(
-                self.logical_id, "Module 'pydantic.v1' has no attribute 'error_wrappers'"
-            ) from e
```

Comment:
This doesn't work because the `AttributeError` isn't thrown by the try block but by `except pydantic.error_wrappers.ValidationError as e:`. So if we keep `except pydantic.error_wrappers.ValidationError as e:` there its going to throw an error that won't be caught

Reply:
Oops I think copy & paste and indentation messed up with the code I had above. I meant to have it as nested try except where we try and except `AttributeError` in the outer try-catch and the inner try-catch would be the original code with 
```
except pydantic.error_wrappers.ValidationError as e:
            error_properties: str = ""
            with suppress(KeyError):
                error_properties = ".".join(str(x) for x in e.errors()[0]["loc"])
            raise InvalidResourceException(self.logical_id, f"Property '{error_properties}' is invalid.") from e
```